### DISCUSSION TITLE
Insert capture at the end of an existing line

### DISCUSSION THREAD

**Discussion by davidmickelsen** [chhoumann/quickadd#505] [Q&A] [answered]
Is it possible to use "Insert After" to insert a capture at the end of an existing line, rather than on a new line below the existing line?

**Comment by nicobako** [comment]
With a capture, you have the option to "insert at the end of the file". If you leave that unselected (and leave "insert after" unselected), then the text will be inserted at the current cursor position.

In your case, that's probably what I'd do: open the not in edit mode, place your cursor in the right place, and run the capture.

Note, QuickAdd offers a lot of different ways to solve this problem, so if you want any other ideas let me know.

One more note, if you're trying to modify the value of a data field you might want to take a look at MetaEdit. The QuickAdd docs has [examples of calling MetaEdit from QuickAdd](https://quickadd.obsidian.guide/docs/Examples/Macro_ChangePropertyInDailyNotes).

**Comment by davidmickelsen** [comment]
Hi Nico, thanks for your reply. What I'm looking for in this case is a capture that will insert text into a file without that file being open, at the end of a line where there is already other text. For example, I might have a file where I record daily health information, and that file might have a line in the frontmatter to record my weight each day:

`weight: `

What I'm looking for is a capture that I can run to input a weight measurement at the end of that line, from anywhere in my vault, without having to open the file. The end result would look like this:

`weight: 62.7`

Sounds like maybe I need to look into MetaEdit.

**Comment by tanjingdong** [comment]
so hard……

**Comment by chhoumann** [comment, marked as answer]
Coming back to this very late - sorry about that. The answer has changed since 2023: Capture can do exactly this now, and MetaEdit has been archived, so here's the current way.

For your weight example, set up the Capture choice like this:

- _Capture To_: the health note (format syntax works here, e.g. `Health/{{DATE:YYYY-MM-DD}}.md`)
- _Write position_: **After line**, with target `weight:`
- Enable **Inline insertion** - the capture lands on the same line, immediately after the matched text, instead of on a new line below
- _Capture format_: ` {{VALUE:weight}}` (leading space so the value sits after the colon)

Run it from anywhere in the vault and `weight: ` becomes `weight: 62.7` without the file ever opening. If you might run it twice on the same line, there's also a **Replace existing value** toggle that overwrites the rest of the line instead of appending.

Shipped in v2.10.0. Docs: https://quickadd.obsidian.guide/docs/Choices/CaptureChoice#insert-after

Closing as resolved.
